### PR TITLE
Update Dalamud.Game.ClientState.Objects.Enums.StatusFlags for 6.0

### PR DESCRIPTION
**Pull request by SheepGoMeh**

### CODE REVIEW

File: Dalamud/Game/ClientState/Objects/Enums/StatusFlags.cs [hunk 1]
```diff
@@ -34,23 +34,23 @@ public enum StatusFlags : byte
...
+        /// Character is a alliance member.
         /// </summary>
-        Friend = 64,
+        AllianceMember = 64,
 
         /// <summary>
-        /// Character is casting.
+        /// Character is in friend list.
         /// </summary>
-        IsCasting = 128,
```
Comment (goaaats) [inline Dalamud/Game/ClientState/Objects/Enums/StatusFlags.cs:54]: Why is IsCasting gone? Is it crafting instead now?
This would break the API.

**Review by goaaats — CHANGES_REQUESTED** [review]

File: Dalamud/Game/ClientState/Objects/Enums/StatusFlags.cs (hunk 1 above)
Comment (SheepGoMeh) [reply Dalamud/Game/ClientState/Objects/Enums/StatusFlags.cs:54]: For whatever odd reason, it seems gone, and everything from PartyMember onwards has been shifted.
What was PartyMember was lighting up when people were crafting, so assuming it's that.

**Comment by UnknownX7** [comment]
OffhandOut is something else too, seems to become true the moment you gain WeaponOut and never returns to false until you change zones (or the character is reloaded). Additionally, I think PartyMember and AllianceMember are only true for people in your party/alliance now.

File: Dalamud/Game/ClientState/Objects/Enums/StatusFlags.cs (hunk 1 above)
Comment (WesselKuipers) [reply Dalamud/Game/ClientState/Objects/Enums/StatusFlags.cs:54]: I was looking at this just now and noticed that one of my friends had the flag of `IsCasting` and nothing else set to true. It looks like everything got shifted?

**Pull request closed without merging**